pkg/varlinkapi/attach.go

```diff
@@ -62,10 +66,10 @@ func (i *LibpodAPI) Attach(call iopodman.VarlinkCall, name string, detachKeys st
 		}
 	}()
 
-	if start {
-		finalErr = startAndAttach(ctr, streams, detachKeys, resize, errChan)
-	} else {
+	if state == libpod.ContainerStateRunning {
 		finalErr = attach(ctr, streams, detachKeys, resize, errChan)
+	} else {
+		finalErr = startAndAttach(ctr, streams, detachKeys, resize, errChan)
```

Comment:
I think you might have broken `Attach` here, which is never supposed to start the container

Reply:
I don't think this is the right fix here. Attach has a `start` parameter and we're ignoring it.

Logic should probably be
`if start && state != libpod.ContainerStateRunning { startAndAttach(...) } else { attach(...) }`

Reply:
The logic here does work, though it's a bit confusing.